ngx-fudis/projects/ngx-fudis/src/lib/components/pagination/pagination.component.ts

```diff
@@ -0,0 +1,342 @@
+   */
+  @Input() pageHref: (index: number) => string = (i) => `#${i + 1}`;
+
+  /**
+   * Internal input for setting the number of pages shown in each side of the current page
+   */
+  @Input() set siblingCount(value: number) {
+    this._siblingCount.set(value);
+  }
+
+  /**
+   * PageChange Emitter
+   */
+  @Output() pageChange = new EventEmitter<number>();
+
+  /**
+   * Page index has been changed
+   */
+  protected hasUserChangedPage = false;
+
+  /**
+   * Protected signal for the total amount of pages
+   */
+  protected _pageCount = signal(0);
+
+  /**
+   * Protected signal for the current page index
+   */
+  protected _pageIndex = signal(0);
+
+  /**
+   * Prefix for aria-label from Fudis translation keys
+   */
+  protected _paginationPrefix = new BehaviorSubject<string>('');
+
+  /**
+   * Current page aria-label from Fudis translation keys
+   */
+  protected _paginationCurrent = new BehaviorSubject<string>('');
+
+  /**
+   * Last page aria-label from Fudis translation keys
+   */
+  protected _paginationLast = new BehaviorSubject<string>('');
+
+  /**
+   * Previous button label from Fudis translation keys
+   */
+  protected _paginationPreviousButton = new BehaviorSubject<string>('');
+
+  /**
+   * Next button label from Fudis translation keys
+   */
+  protected _paginationNextButton = new BehaviorSubject<string>('');
+
+  /**
+   * Previous button aria-label from Fudis translation keys
+   */
+  protected _paginationPreviousButtonAria = new BehaviorSubject<string>('');
+
+  /**
+   * Next button aria-label from Fudis translation keys
+   */
+  protected _paginationNextButtonAria = new BehaviorSubject<string>('');
+
+  /**
+   * Main CSS classes for buttons
+   */
+  protected _mainCssPrevClass: BehaviorSubject<string> = new BehaviorSubject<string>(
+    'fudis-pagination-button__previous',
+  );
+  protected _mainCssNextClass: BehaviorSubject<string> = new BehaviorSubject<string>(
+    'fudis-pagination-button__next',
+  );
+
+  /**
+   * Returns an array of numbers from given starting to ending number
+   */
+  protected range = (start: number, end: number): number[] => {
+    const length = end - start + 1;
+    return Array.from({ length }, (_, i) => start + i);
+  };
+
+  /**
+   * HTML id
+   */
+  protected _id: string;
+
+  /**
+   * Private signal for counting Pagination item visible siblings
+   */
+  private _siblingCount = signal(2);
+
+  /**
+   * After DOM is updated, restore focus to the new active element
+   */
+  private userSelectedIndex = -1;
+
+  /**
+   * Stores the previous page index for change detection
+   */
+  private prevPageIndex: number | undefined;
+
+  /**
+   * To observe container size changes and adjust Pagination sibling count based on viewport changes
+   */
+  private observer?: ResizeObserver;
+
+  /**
+   * Computed list of pagination items (page numbers + ellipses)\
+   * Recalculated whenever page count, current page, or sibling count changes.
+   */
+  itemList = computed(() =>
```

Comment:
Our first computed signal in the whole library, yey :D But should/could this also be set as `protected` like any other variables?